ui/src/adapters/api/credentials.ts

```diff
@@ -396,138 +457,103 @@ export async function createLink({
-  linkDetail: { proofTypes: ProofTypeInput[]; schemaType: string };
-};
-
-export type AuthQRCode = {
+export type AuthMessage = {
+  deepLink: string;
   linkDetail: { proofTypes: ProofType[]; schemaType: string };
-  qrCodeLink: string;
   qrCodeRaw: string;
-  sessionID: string;
+  universalLink: string;
 };
 
-const authQRCodeParser = getStrictParser<AuthQRCodeInput, AuthQRCode>()(
+const authMessageParser = getStrictParser<AuthMessage>()(
   z.object({
-    linkDetail: z.object({ proofTypes: proofTypeParser, schemaType: z.string() }),
-    qrCodeLink: z.string(),
+    deepLink: z.string(),
+    linkDetail: z.object({
+      proofTypes: z.array(z.nativeEnum(ProofType)),
+      schemaType: z.string(),
+    }),
     qrCodeRaw: z.string(),
-    sessionID: z.string(),
+    universalLink: z.string(),
   })
 );
 
-export async function createAuthQRCode({
+export async function createAuthMessage({
   env,
+  identifier,
   linkID,
   signal,
 }: {
   env: Env;
+  identifier: string;
   linkID: string;
   signal?: AbortSignal;
-}): Promise<Response<AuthQRCode>> {
+}): Promise<Response<AuthMessage>> {
   try {
     const response = await axios({
       baseURL: env.api.url,
       method: "POST",
       signal,
-      url: `${API_VERSION}/credentials/links/${linkID}/qrcode`,
+      url: `${API_VERSION}/identities/${identifier}/credentials/links/${linkID}/qrcode`,
     });
-    return buildSuccessResponse(authQRCodeParser.parse(response.data));
+    return buildSuccessResponse(authMessageParser.parse(response.data));
   } catch (error) {
     return buildErrorResponse(error);
   }
 }
 
-type IssuedQRCodeInput = {
-  qrCodeLink: string;
-  schemaType: string;
-};
-
-const issuedQRCodeParser = getStrictParser<IssuedQRCodeInput, IssuedQRCode>()(
-  z
-    .object({
-      qrCodeLink: z.string(),
-      schemaType: z.string(),
-    })
-    .transform(({ qrCodeLink, schemaType }) => ({ qrCode: qrCodeLink, schemaType: schemaType }))
+const issuedMessageParser = getStrictParser<IssuedMessage>()(
+  z.object({
+    schemaType: z.string(),
+    universalLink: z.string(),
+  })
 );
 
-export async function getIssuedQRCodes({
+export async function getIssuedMessages({
   credentialID,
   env,
+  identifier,
   signal,
 }: {
   credentialID: string;
   env: Env;
-  signal: AbortSignal;
-}): Promise<Response<[IssuedQRCode, IssuedQRCode]>> {
+  identifier: string;
+  signal?: AbortSignal;
+}): Promise<Response<[IssuedMessage, IssuedMessage]>> {
   try {
-    const [qrLinkResponse, qrRawResponse] = await Promise.all([
+    const [universalLinkResponse, deepLinkResponse] = await Promise.all([
       axios({
         baseURL: env.api.url,
+        headers: {
+          Authorization: buildAuthorizationHeader(env),
+        },
         method: "GET",
-        params: { type: "link" },
         signal,
-        url: `${API_VERSION}/credentials/${credentialID}/qrcode`,
+        url: `${API_VERSION}/identities/${identifier}/credentials/${credentialID}/qrcode`,
```

Comment:
@x1m3 Are we still keeping "qrcode" in the url? We should rename it to something else.

Reply:
maybe `.../fetchMessage` and also add the message itself there? WDYT @demonsh?